src/lib/Microsoft.Health.Fhir.Ingest.Template/MeasurementExtractor.cs

```diff
@@ -126,15 +126,23 @@ protected virtual IEnumerable<JToken> MatchTypeTokens(JObject token)
             EnsureArg.IsNotNull(token, nameof(token));
             var evaluator = CreateRequiredExpressionEvaluator(Template.TypeMatchExpression, nameof(Template.TypeMatchExpression));
 
+            JObject tokenClone = null;
+
             foreach (var extractedToken in evaluator.SelectTokens(token))
             {
                 // Add the extracted data as an element of the original data.
                 // This allows subsequent expressions access to data from the original event data
+                if (tokenClone == null)
+                {
+                    tokenClone = new JObject(token);
+                }
 
-                var tokenClone = token.DeepClone() as JObject;
                 tokenClone.Add(MatchedToken, extractedToken);
                 yield return tokenClone;
+                tokenClone.Remove(MatchedToken);
```

Comment:
Does this work?  We are returning the tokenClone and them updating the same reference to remove the property.  I don't think we can do the remove.  Anything downstream of this will potentially be missing the MatchedToken.

Reply:
This should work as intended as this is a synchronous `IEnumerable`. We first `yield` the cloned `JToken` and then produce a `Measurement` from it. Finally we return here to remove the `MatchedToken` property and start the loop again. The code producing the `Measurement` does not hold onto a reference for the token it was given. 

We have various unit tests which should be testing out this scenario, such as: https://github.com/microsoft/iomt-fhir/blob/main/test/Microsoft.Health.Fhir.Ingest.Template.UnitTests/CalculatedFunctionContentTemplateTests.cs#L79